### PR TITLE
Cross-platform client for Python host

### PR DESCRIPTION
**Pull request by cgranade**
This PR completes the client side of the host support tracked in #31. The README needs a bit of cleanup once the corresponding documentation has been added to quantum-docs-pr, but this should otherwise be good to go.

### CODE REVIEW

File: Interoperability/python/README.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+```
+
+On top of simulation, you can also do quantum resources estimation including 
+the count of primitive operations used by the algorithm and the number of required qubits.
+For this, invoke the `trace` method on the operation:
+```python
+r = HelloAgain.trace(count=5, name="Counting")
+```
+
+You may use the `qsharp.print_tracer_counts` method to print the trace results to the console:
```
Comment (anpaz-msft) [inline Interoperability/python/README.md]: This is not the case anymore, the results contain already a dictionary with the resources calculated.

File: Interoperability/python/README.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+```
+
+
+You can now also compile Q# operations on the fly from Python
+and simulate them.
+To create an operation on the fly, the `qsharp` module exports a `compile` method which receives two parmaters
+* id: a unique identifier of this snippet.
+* source: a valid Q# code snippet with the operation definition, for example:
+```python
+hello = qsharp.compile("snippet_1", """
```
Comment (anpaz-msft) [inline Interoperability/python/README.md]: the id parameter is now gone.

File: Interoperability/python/src/Operations.qs [hunk 3]
```diff
@@ -0,0 +1,36 @@
+﻿/// Q# code should be in one or more .qs files that live
```
Comment (anpaz-msft) [inline Interoperability/python/src/Operations.qs]: this file should not go here, right?

File: Interoperability/python/src/driver.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
+#!/bin/env python
```
Comment (anpaz-msft) [inline Interoperability/python/src/driver.py]: similar this file should not be here.

File: Interoperability/python/src/qsharp/__init__.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    ops = [
+        QSharpCallable(op, "snippets")
+        for op in client.compile(code)
+    ]
+    if len(ops) == 1:
+        return ops[0]
+
+def reload() -> None:
+    """
+    Reloads the current IQ# workspace, discarding any current snippets and
```
Comment (anpaz-msft) [inline Interoperability/python/src/qsharp/__init__.py]: I don't think we're discarding current snippets (yet).

**Comment by anpaz-msft** [comment]
Because you removed `step-build-python.yml` you'll need to update `build.yml`.

File: Interoperability/python/src/Operations.qs (hunk 3 above)
Comment (cgranade) [reply Interoperability/python/src/Operations.qs]: Good catch, moving this and `driver.py` to the samples repo.

File: Interoperability/python/src/qsharp/__init__.py (hunk 5 above)
Comment (cgranade) [reply Interoperability/python/src/qsharp/__init__.py]: Good catch, I've fixed the docstring accordingly.

File: Interoperability/python/README.md (hunk 2 above)
Comment (cgranade) [reply Interoperability/python/README.md]: Honestly, most of the README can be deferred to the docs now, I think. I'll take a pass to focus it on what makes the most sense here.

File: Interoperability/python/README.md (hunk 1 above)
Comment (cgranade) [reply Interoperability/python/README.md]: Resolved by other changes to README. Thanks!

**Comment by cgranade** [comment]
> Because you removed `step-build-python.yml` you'll need to update `build.yml`.

Ah, good point. I thought I'd done so, but must have missed it, sorry.

**Pull request merged**